### DISCUSSION TITLE
Is possible in config override default settings?

### DISCUSSION THREAD

**Discussion by damanis** [linrongbin16/fzfx.nvim#423] [General]
@linrongbin16 
For example, I want to change vertical preview layout to horizontal ("top,75%", for example) for LiveGrep. I tried override it in fzfx.setup call, but without success (I got horizontal layout, but LiveGrep didn't work). Currently, I made changes in git clone of fzfx.

**Comment by linrongbin16** [comment]
hi @damanis, you can set basic fzf opts here: https://github.com/linrongbin16/fzfx.nvim/blob/6ea6951599f355e4f01ebac7d4eaa033777251ea/lua/fzfx/config.lua#L4370.

the `fzf_opts` option in root level will apply to all commands.

but it has the lowest priority, could be overwrite by the `fzf_opts` in specific commands.

but it should be ok, because all the `fzf_opts` in all the commands didn't specify `top`.

**Comment by damanis** [comment]
I try this, but it does not work.
```
  local fzfx = require("fzfx")
  fzfx.setup({
    fzf_opts = {
        { "--preview-window", "top,75%,+{2}-/2" },
    },
  })
```

**Reply by linrongbin16** [reply]
I will check later.

**Reply by linrongbin16** [reply]
hi @damanis, even though I haven't do testing, but in my memory, the root level `fzf_opts` and commands level `fzf_opts` is processed in below order:

for example:
```
{
  live_grep = {
    fzf_opts = {
      '--no-multi',
      { '--preview-window', '+{2}-/2'},
    },
  },
  fzf_opts = {
    '--multi',
    { '--preview-window', 'top,75%' },
  },
}
```

current engine will use commands fzf opts to overwrite root level fzf opts, finally emit below fzf opts to the `fzf` binary:

```
fzf --no-multi --preview-window '+{2}-/2'
```

that is the reason of root level cannot work, because command level fzf opts already has a `--preview-window`, and has higher priority.

maybe I could provide another root level option `final_fzf_opts`, so you can configure it like:

```
{
  live_grep = {
    fzf_opts = {
      '--no-multi',
      { '--preview-window', '+{2}-/2'},
    },
  },
  fzf_opts = {
    '--multi',
  },
  final_fzf_opts = {
    { '--preview-window', 'top,75%' },
  },
}
```

the `final_fzf_opts` will have even higher priority that can overwrite command level fzf opts, so let you config the global option.

**Comment by damanis** [comment]
@linrongbin16 
Maybe, better `user_fzf_opts`.

**Reply by linrongbin16** [reply]
I feel like all the options can be configured by user, the word `user` here may make user think other `fzf_opts` are not for them.

so maybe let's use another word? I think the most different is the priority.

**Comment by damanis** [comment]
As an option `override_fzf_opts`.

**Reply by linrongbin16** [reply]
ok, I will implement this one

**Reply by linrongbin16** [reply]
hi @damanis , I merged https://github.com/linrongbin16/fzfx.nvim/pull/427 into main branch (also see https://github.com/linrongbin16/fzfx.nvim/pull/427#issuecomment-1823765807), please have a try!

**Comment by damanis** [comment]
@linrongbin16. It works!

**Reply by savchenko** [reply]
Could you mark this one as resolved / close the thread?
